### PR TITLE
fix: error format in check command

### LINKED ISSUE #1332: Invalid Error Format
`start` command throws an error in an unexpected format
**Input**
```graphql
schema @upstream(httpCache: "true") {
  query: Query
}
```

**Actual**

```
 INFO File read: ./examples/rest-api.graphql ... ok
ERROR Validation Error
• Parsing failed because of invalid type: string "true", expected a boolean [@upstream, httpCache]
```

**Expected**

```
 INFO File read: ./examples/rest-api.graphql ... ok
ERROR Invalid Configuration
Caused by:
  • Parsing failed because of invalid type: string "true", expected a boolean [at schema.@upstream.httpCache]
```

### PR DESCRIPTION
**Pull request by Bhavyajain21**
Fixes #1332 
/claim #1332

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [x] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

## Summary by CodeRabbit

- **New Features**
	- Introduced a new schema directive for improved caching behavior in API requests.

- **Bug Fixes**
	- Updated error messages to be more informative, including better formatting and additional schema details.

### CODE REVIEW

File: examples/rest-api.graphql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+schema @upstream(httpCache: "true") {
+  query: Query
+}
```
Comment (tusharmath) [inline examples/rest-api.graphql:3]: this file isn't required.

Comment (Bhavyajain21) [reply examples/rest-api.graphql]: removed the file

**Comment by Bhavyajain21** [comment]
image

Adding screenshot

File: src/valid/error.rs [hunk 2]
```diff
@@ -9,20 +9,20 @@ pub struct ValidationError<E>(Vec<Cause<E>>);
...
                     &(format!(
-                        " [{}]",
+                        " [at schema.{}]",
                         error
                             .trace
                             .iter()
                             .cloned()
                             .collect::<Vec<String>>()
-                            .join(", ")
+                            .join(".")
```
Comment (tusharmath) [inline src/valid/error.rs:25]: I think you got the problem! We are wrapping ValidationError in a anyhow::Error somewhere, which forcefully converts the ValidationError into a String using this Display implementation, and then wraps it inside of anyhow::Error.

The solution should make sure that we don't convert a ValidationError into a String. Infact we should drop the Display implementation of ValidationError.

Comment (Bhavyajain21) [reply src/valid/error.rs:25]: Let me check

**Comment by tusharmath** [comment]
Closing this @Bhavyajain21 in favour of https://github.com/tailcallhq/tailcall/pull/1335/files

**Pull request closed without merging**